plonk/src/verifier.rs

```diff
@@ -0,0 +1,217 @@
+use std::ops::Mul;
+use ark_bls12_381::{Bls12_381, Fr};
+use ark_ec::PairingEngine;
+use ark_poly::{EvaluationDomain, Polynomial};
+use kzg::{KzgCommitment, KzgScheme};
+use crate::CompiledCircuit;
+use crate::challenge::ChallengeParse;
+use crate::prover::Proof;
+
+impl CompiledCircuit {
+    pub fn verify(&self, proof: Proof) -> bool {
+        println!("Verifying...");
+
+        #[cfg(test)]
+        println!("Precompute");
+        let (q_m_c, q_l_c, q_r_c, q_o_c, q_c_c, ssigma_1_c, ssigma_2_c, ssigma_3_c) = self.get_circuit_commitment();
+
+        #[cfg(test)]
+        println!("Verify challenges");
+
+        let (alpha, beta, gamma, evaluation_challenge, v, u) = self.verify_challenges(&proof);
+
+        assert_eq!(u, proof.u, "Verify: Challenge verification failed.");

```

Comment:
Shouldn't panic in a lib. Also, because the protocol is ZK, the only result it should give is [0;1] or [true; false] therefor, return false here should be a better alternative.